common/web/sentry-manager/src/index.ts

```diff
@@ -1,250 +1,263 @@
-          errorMsg = new Error(
-            args.reduce(function(accumulator, currentValue) {
-              return accumulator.toString() + ' ' + currentValue.toString();
-            }, '')
-          );
-        }
-        return errorMsg;
+    let oldConsoleWarn = console.warn;
+    console.warn = reportingConsoleWarn; // defined via function hoisting
+    function reportingConsoleWarn() {
+      let args = Array.prototype.slice.call(arguments);
+      if(_this._enabled) {
+        // @ts-ignore   // should be an enum with the same value... if we could import it.
+        Sentry.captureMessage(reduceConsoleArgs(args), { level: 'warning' });
       }
+      return oldConsoleWarn.apply(console, args);
     }
 
-    init() {
-      // Do the actual Sentry initialization.
-      //@ts-ignore
-      Sentry.init({
-        beforeSend: this.prepareEventDebugWrapper.bind(this),
-        // Not using beforeBreadcrumb because that caused breadcrumbs to get lost in Sentry
-        debug: DEBUG,
-        dsn: 'https://cf96f32d107c4286ab2fd82af49c4d3b@o1005580.ingest.sentry.io/5983524', // keyman-web DSN
-        release: com.keyman.KEYMAN_VERSION.VERSION_GIT_TAG,
-        environment: com.keyman.KEYMAN_VERSION.VERSION_ENVIRONMENT,
-      });
-
-      this.initConsole();
+    function reduceConsoleArgs(args) {
+      let errorMsg = args[0];
+      // Make sure errorMsg is either an error or string.
+      // It's therefore best to pass in new Error('msg') instead of just 'msg' since
+      // that'll give you a stack trace leading up to the creation of that new Error
+      // whereas if you just pass in a plain string 'msg', the stack trace will include
+      // reportingConsoleError and reportingConsoleCall
+      if (!(errorMsg instanceof Error)) {
+        // stringify all args as a new Error (which creates a stack trace)
+        errorMsg = new Error(
+          args.reduce(function(accumulator, currentValue) {
+            return accumulator.toString() + ' ' + currentValue.toString();
+          }, '')
+        );
+      }
+      return errorMsg;
     }
+  }
 
-    get enabled(): boolean {
-      return this._enabled;
-    }
+  init() {
+    // Do the actual Sentry initialization.
+    //@ts-ignore
+    Sentry.init({
+      beforeSend: this.prepareEventDebugWrapper.bind(this),
+      // Not using beforeBreadcrumb because that caused breadcrumbs to get lost in Sentry
+      debug: DEBUG,
+      dsn: 'https://cf96f32d107c4286ab2fd82af49c4d3b@o1005580.ingest.sentry.io/5983524', // keyman-web DSN
+      release: KEYMAN_VERSION.VERSION_GIT_TAG,
+      environment: KEYMAN_VERSION.VERSION_ENVIRONMENT,
+    });
 
-    set enabled(value: boolean) {
-      this._enabled = value;
-    }
+    this.initConsole();
+  }
+
+  get enabled(): boolean {
+    return this._enabled;
+  }
+
+  set enabled(value: boolean) {
+    this._enabled = value;
   }
 }
+
+// Publish to the window.
+window['KeymanSentryManager'] = KeymanSentryManager;
```

Comment:
Is this a new global? It's used only in Android Engine and iOS Engine, right?

Reply:
Yes and yes.  I made sure to double-check that detail.

As we're getting rid of `com` and related namespacing structure, it couldn't be referenced in quite the same way as before.